pages/learn.vue

```diff
@@ -149,23 +169,74 @@ export default class extends QiskitPage {
   learnLevels = LEARN_LEVEL_OPTIONS
   timeScales = TIME_SCALE_OPTIONS
 
+  beforeRouteEnter (route, _, next) {
+    next(learnPage => learnPage._parseFilterFromUrl(route))
+  }
+
+  beforeRouteUpdate (route, _, next) {
+    this._parseFilterFromUrl(route)
+    next()
+  }
+
+  _parseFilterFromUrl (route) {
+    const timeScale = route.query.timeScale || TIME_SCALES.any
+    const learnLevel = route.query.learnLevel || LEARN_LEVELS.all
+    this.$store.commit('setTimeScale', timeScale)
+    this.$store.commit('setLearnLevel', learnLevel)
+  }
+
   setTimeScale (scale: TimeScale): void {
-    this.$store.commit('setTimeScale', scale)
+    const { timeScale: currentScale } = this as any
+    if (currentScale === scale) {
+      return
+    }
+    this._updateQueryParameter('timeScale', scale)
   }
 
   setLearnLevel (tabIndex: number) {
+    const { learnLevel: currentLevel } = this as any
     const level = this.learnLevels[tabIndex]
-    this.$store.commit('setLearnLevel', level)
+    if (currentLevel === level) {
+      return
+    }
+    this._updateQueryParameter('learnLevel', level)
+  }
+
+  _updateQueryParameter (paramName: string, value: string) {
+    this.$router.push({
+      query: {
+        ...this.$route.query,
+        [paramName]: value
+      }
+    })
   }
 
-  get isAMinuteForBeginner (): boolean {
-    return (this as any).timeScale === TIME_SCALES.minute &&
-      (this as any).learnLevel === LEARN_LEVELS.beginner
+  get showingEverything (): boolean {
+    const { timeScale, learnLevel } = (this as any)
+    return timeScale === TIME_SCALES.any && learnLevel === LEARN_LEVELS.all
   }
 
-  get isAMinuteForAdvanced (): boolean {
-    return (this as any).timeScale === TIME_SCALES.minute &&
-      (this as any).learnLevel === LEARN_LEVELS.advanced
+  get showingOneMinuteForBeginner (): boolean {
+    return this.showingOneMinute && this.showingLevel(LEARN_LEVELS.beginner)
+  }
+
+  get showingOneMinuteForAdvanced (): boolean {
+    return this.showingOneMinute && this.showingLevel(LEARN_LEVELS.advanced)
+  }
+
+  get showingOneMinute (): boolean {
+    const { timeScale } = (this as any)
+    return [TIME_SCALES.any, TIME_SCALES.minute].includes(timeScale)
+  }
+
+  showingLevel (level: LearnLevel): boolean {
```

Comment:
Are you missing the `get` here?

Reply:
No, all others are properties but this one is a function that recieves the learn level you're checking.